library/src/main/java/org/apache/apex/malhar/lib/utils/serde/SerdeCollectionSlice.java

```diff
@@ -37,35 +37,40 @@
  * @since 3.5.0
  */
 @InterfaceStability.Evolving
-public class SerdeListSlice<T> implements Serde<List<T>, Slice>
+public class SerdeCollectionSlice<T, CollectionT extends Collection<T>> implements Serde<CollectionT, Slice>
```

Comment:
The serde is for any collection, not just a List. That's why I made this change, since I need a serde for a Set and the same serde works for a Set as well.